### PR TITLE
introduce pack-single-json osqueryi argument to output valid json

### PR DESCRIPTION
**Pull request by jspaleta**
Introduces new output mode useful when running packs from osqueryi.
This change attempts to address issue #4432 

new output mode creates valid json hash with pack name, and query names.
Parsing example using jq:
`osqueryi --pack osquery-monitoring --pack_single_json|jq`

### CODE REVIEW

File: osquery/devtools/shell.cpp [hunk 1]
```diff
@@ -56,6 +56,7 @@ namespace osquery {
 /// Define flags used by the shell. They are parsed by the drop-in shell.
 SHELL_FLAG(bool, csv, false, "Set output mode to 'csv'");
 SHELL_FLAG(bool, json, false, "Set output mode to 'json'");
+SHELL_FLAG(bool, pack_single_json, false, "Set output mode to 'pack_single_json'");
```
Comment (theopolis) [inline osquery/devtools/shell.cpp]: nitpick, this is a long flag, can `--mjson` stand for "merged JSON" or `--sjson` for "single JSON" work?

Comment (theopolis) [reply osquery/devtools/shell.cpp]: In any case I think the flag description needs a detail about what this does, the others are more obvious, hehe!

**Comment by theopolis** [comment]
ok to test

**Comment by jspaleta** [comment]
hey @fmanco,

Thanks for the feedback.  
Here's my refactor plan:
1) build a pack JSON object in shell_exec and grow the object as part of the shell execution loop.
2) serialize object at the end execution.

File: osquery/devtools/shell.cpp (hunk 1 above)
Comment (jspaleta) [reply osquery/devtools/shell.cpp]: hey @theopolis ,
I'm happy with any long flag name that makes the most sense.  I'll refactor to using --mjson on next commit when i add the more detailed description.

**Comment by fmanco** [comment]
Hey @jspaleta yeah you need to accumulate the results and serialize at the end. I don't think this is a trivial refactor though, could you explain how you are going to implement?

**Comment by fmanco** [comment]
ping @jspaleta

**Comment by jspaleta** [comment]
@fmanco   wow sorry.
I actually wrote up a plan in a comment and thought i had finished the comment.  Lost forever I guess.

I have a plan!  
I was going to go back and refactor by:
1)  having  a dedicated  pack_json object with JSON::newObject();  in shell.cpp 

2) append JSON representation of query results into the pack_json object ias part of the query loop in shell.cpp.

3) have a dedicated packJsonNPrint function taking the pack_json object as argument in print.cpp
It will step into the pack_json object and iterate accordingly. It will reuse some of the logic from jsonPrint when it needs to print the query results.

4) call packJsonPrint function at end of shell.cpp query processing.  

I've got time end of this week, this weekend to refactor along those lines.

**Comment by fmanco** [comment]
@jspaleta Sounds like a good plan. Thanks!

**Comment by fmanco** [comment]
In an effort to keep our open PR list clean and easily actionable we're closing stale pull requests. If you plan to work on this feature in the future simply re-open the PR when you have updates and we're happy to follow up. Check our updated [contributing](https://github.com/facebook/osquery/blob/master/CONTRIBUTING.md) guidelines or comment if have questions. Thanks.

**Pull request closed without merging**